vmware-base/src/main/java/com/cloud/hypervisor/vmware/mo/DatastoreMO.java

```diff
@@ -416,22 +414,39 @@ public String searchFileInSubFolders(String fileName, boolean caseInsensitive, S
             s_logger.error(msg);
             throw new CloudException(msg);
         }
+        String parentFolderPath;
+        String absoluteFileName = null;
         for (HostDatastoreBrowserSearchResults result : results) {
             List<FileInfo> info = result.getFile();
             if (info != null && info.size() > 0) {
                 for (FileInfo fi : info) {
-                    absoluteFileName = parentFolderPath = result.getFolderPath();
-                    s_logger.info("Found file " + fileName + " in datastore at " + absoluteFileName);
-                    if (parentFolderPath.endsWith("]"))
-                        absoluteFileName += " ";
-                    absoluteFileName += fi.getPath();
-                    if(isValidCloudStackFolderPath(parentFolderPath, searchExcludedFolders)) {
+                    String possibleFilename = null;
+                    possibleFilename = parentFolderPath = result.getFolderPath();
```

Comment:
should this double assignment really be here? it does not make the code clearer.

Reply:
it's copied/refactored code (nothing new I wrote mostly), see old code line 423:
```
                    absoluteFileName = parentFolderPath = result.getFolderPath();
```